bunsen/bunsen-avro/src/test/resources/r4-custom-schemas/bunsen-test-patient-schema.json

```diff
@@ -562,6 +561,228 @@
                         "doc": "Reference field",
                         "default": null
                       },
+                      {
+                        "name": "identifier",
```

Comment:
As discussed offline, can you please do some minimal memory profiling between the old and new schema and see how much impact this larger schema has on the memory footprint?

Reply:
I did some basic memory profiling between the old and new schema and it looks like there is an increase of roughly by 20% in the size of parquet files in the new schema. This is because of the recent changes where in by default now we capture a nested field size of 2 for repeated fields (earlier it was 1 due to a bug which is fixed in this PR). This reminds us to relook at the `recursiveDepth` parameter and do the right recursive capture. I have created this [issue](https://github.com/google/fhir-data-pipes/issues/1062) to relook into this.

On the other hand, due to the changes in this [PR](https://github.com/google/fhir-data-pipes/pull/1047) to flush the parquet files in the finishBundle method, it has created some regression on the number of parquet files generation for `DirectRunner`. Earlier (before this [PR](https://github.com/google/fhir-data-pipes/pull/1047) changes), the number of parquet files created for each resource type is equal to the `parallelism` parameter being passed in the `FhirEtlOptions`, whereas after the change the number of files created is equal to the number of FHIR resources i.e. for a total of 12K observations, it created 12K parquet files. This was causing an `OutOfMemoryError` while checking number of the records in the parquet files using `parquet-tools` util jar in the e2e tests.

I have created this [issue](https://github.com/google/fhir-data-pipes/issues/1063) for the above regression.

Reply:
I have increased the heap size in the current PR to check the number of FHIR resources in parquet files. This has to be reverted back once this [issue](https://github.com/google/fhir-data-pipes/issues/1063) is fixed.

Interestingly, the e2e-tests was not failing in the PR where the changes were made to fix the parquet file generation for DataFlowRunner. This was because the default max heap size (8gb) was sufficient enough to check the count, but with the recent increase in schema size i.e. ~20%, this step is failing in the current PR.

Reply:
Thanks for checking and the explanation.
For future reference (following our off-PR discussion): I think the 20% increase is for the small dataset we use for e2e test. For larger/more realistic datasets, this ratio should be much smaller, correct?

Reply:
Yes @bashir2, for larger dataset the difference seems to be negligible. I verified this for the 8K Patients dataset and here is the data size of the parquet files before and after the changes.

Before Change
--------------
Patients - 2.7MB
Encounter - 86MB
Observation - 123MB

After Change
-------------
Patients - 3MB
Encounter - 86MB
Observation - 124MB

So, I am assuming that for smaller dataset due to the difference in schema size, the difference is visible, but as the data grows this schema difference seems to be negligible (unless even the data is captured with so many levels)